pkg/controllers/subnetport/addressbinding_webhook.go

```diff
@@ -56,8 +57,24 @@ func (v *AddressBindingValidator) Handle(ctx context.Context, req admission.Requ
 			log.Error(err, "error while decoding AddressBinding", "AddressBinding", req.Namespace+"/"+req.Name)
 			return admission.Errored(http.StatusBadRequest, err)
 		}
-		if !reflect.DeepEqual(ab.Spec, oldAddressBinding.Spec) {
-			return admission.Denied("update AddressBinding is not allowed")
+		if ab.Spec.VMName != oldAddressBinding.Spec.VMName || ab.Spec.InterfaceName != oldAddressBinding.Spec.InterfaceName {
+			return admission.Denied("update AddressBinding vmName/interfaceName is not allowed")
+		}
+	}
+	if ab.Spec.IPAddressAllocationName != "" {
+		ipAllocation := &v1alpha1.IPAddressAllocation{}
+		if err := v.Client.Get(context.TODO(), types.NamespacedName{
+			Namespace: ab.Namespace,
+			Name:      ab.Spec.IPAddressAllocationName,
+		}, ipAllocation); err != nil {
+			log.Error(err, "failed to get IPAddressAllocation", "IPAddressAllocation", ab.Namespace+"/"+ab.Spec.IPAddressAllocationName)
+			return admission.Denied("ipAddressAllocationName is not valid")
+		}
+		if ipAllocation.Spec.IPAddressBlockVisibility != v1alpha1.IPAddressVisibilityExternal {
+			return admission.Denied("ipAddressBlockVisibility of IPAddressAllocation must be \"External\"")
+		}
+		if net.ParseIP(ipAllocation.Status.AllocationIPs) == nil {
```

Comment:
If we check realization status, then we cannot support the case that users create IPAddressAllocation and AddressBinding in one YAML? Maybe let us just check the allocation is for a single IP from the IPAddressAllocation.Spec, but do not check an IP is allocated or not in Status.